examples/gpu/eemumu_AV/SubProcesses/P1_Sigma_sm_epem_mupmum/check.cc

```diff
@@ -160,70 +200,42 @@ int main(int argc, char **argv)
   using mgOnGpu::nparf;
   using mgOnGpu::npar;
   const int nRnarray = np4*nparf*nevt; // (NB: ASA layout with nevt=npagR*neppR events per iteration)
-
-#ifdef __CUDACC__
-  const int nbytesRnarray = nRnarray * sizeof(fptype);
-  fptype* devRnarray = nullptr; // AOSOA[npagR][nparf][np4][neppR] (NB: nevt=npagR*neppR)
-  checkCuda( cudaMalloc( &devRnarray, nbytesRnarray ) );
-#if defined MGONGPU_CURAND_ONHOST or defined MGONGPU_COMMONRAND_ONHOST
-  fptype* hstRnarray = nullptr; // AOSOA[npagR][nparf][np4][neppR] (NB: nevt=npagR*neppR)
-  checkCuda( cudaMallocHost( &hstRnarray, nbytesRnarray ) ); // pinned memory
-#endif
-#else
-  fptype* hstRnarray = nullptr; // AOSOA[npagR][nparf][np4][neppR] (NB: nevt=npagR*neppR)
-  hstRnarray = new fptype[nRnarray]();
-#ifdef MGONGPU_COMMONRAND_ONHOST
-  const int nbytesRnarray = nRnarray * sizeof(fptype);
-#endif
+  auto devRnarray = devMakeUnique( nRnarray ); // AOSOA[npagR][nparf][np4][neppR] (NB: nevt=npagR*neppR)
+#if defined MGONGPU_CURAND_ONHOST or defined MGONGPU_COMMONRAND_ONHOST or not defined __CUDACC__
+  auto hstRnarray = hstMakeUnique( nRnarray ); // AOSOA[npagR][nparf][np4][neppR] (NB: nevt=npagR*neppR)
```

Comment:
Here I would still make the explicit distinction between hstMakeUnique in c++ and hstMakeUniquePinned in CUDA

Reply:
I take this back as discussed above